### PR TITLE
Document Storybook release process

### PR DESCRIPTION
**Pull request by shilman**
Issue: N/A

## What I did

Document our release/decision-making process initially for discussion purposes, and finally as a reference so we have some standards about this for our releases.

- Gathered/incorporated some feedback from this gist: https://gist.github.com/shilman/7993bc0c74974ef1ace7c4f96900554c
- Extended it to incorporate the concept of blocking bugs as discussed with @tmeasday 

## How to test

N/A

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+## Blocking bugs
+
+Most PATCH releases come from community members, who generously fix problems as
+they come up. However, there are also bugs that never get picked up and just sit
+there gathering upvotes and "me too" comments. We need a way to make sure that
+these bugs get addressed.
+
+For every non-PATCH release, we nominate a small number of bugs that must be
+addressed before a release can go out by adding them to the milestone. For example, here's a list of blocking bugs [for the 3.2 milestone](https://github.com/storybooks/storybook/milestone/3).
```
Comment (tmeasday) [inline RELEASES.md]: Is this link broken?

File: RELEASES.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+Adding bugs to the milestone helps people looking for good ways to contribute,
+or to understand what is blocking the release so they can actually do something
+about it. Discussion about which bugs are critical happens in the `#maintenance`
+channel [in our Slack](https://storybooks-slackin.herokuapp.com/).
+
+If you're experiencing a bug, the best way to make sure that it gets attention
+is to upvote it by adding a "thumbs-up" reaction in Github. This way important
+bugs quickly bubble to the top [with a
+search](https://github.com/storybooks/storybook/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).
```
Comment (tmeasday) [inline RELEASES.md]: Can we change this link to filter out non-`bug`s, `fixed` bugs, bugs with a milestone, etc

File: RELEASES.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+And of course, the best way to make sure a bug gets addressed quickly is to fix
+it yourself and issue a PR. If the fix is good, we'll try to release it quickly
+in a patch release.
+
+## Decision-making
+
+-   For PATCH changes, all discussion can occur in issue/PR comments (and random slack chat as needed).
+-   For MINOR feature changes, there are multiple stages of discussion:
+    -   The feature may be discussed in an issue before it is implemented (ex: <https://github.com/storybooks/storybook/issues/151>)
+    -   Once it's implemented, the discussion may be occur on the PR (ex: <https://github.com/storybooks/storybook/issues/151>)
```
Comment (tmeasday) [inline RELEASES.md]: Should this be a PR link?

**Review by tmeasday — CHANGES_REQUESTED** [review]

**Comment by shilman** [comment]
@tmeasday thanks for catching the typos. added a `bug` filter to the issue search, but did not filter out issues that have a milestone because it's useful to see when things are scheduled among the results. can filter more later if things get too out of control.

**Review by tmeasday — APPROVED** [review]

**Comment by tmeasday** [comment]
LGTM! Might want to wait for a couple more approvals to this process change.

File: RELEASES.md (hunk 1 above)
Comment (shilman) [inline RELEASES.md]: It's all open issues in the 3.2 milestone, but I think that's OK?

Comment (tmeasday) [inline RELEASES.md]: Yeah I was wrong, I just eyeballed and though it was supposed to say `..../3.2`

**Review by igor-dv — APPROVED** [review]
LGTM

File: RELEASES.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
+# Release Process
+
+A Storybook release process based on [Semver](http://semver.org/). In short:
+1\. merge bugfix PRs directly into master automatically release (~daily)
```
Comment (alexandrebodin) [inline RELEASES.md]: I would end the lines here for clarity

Comment (shilman) [inline RELEASES.md]: @alexandrebodin thanks. some weird linting auto-fix 😄

**Review by usulpro — APPROVED** [review]

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
Looks good to me!

**Pull request merged**